nomad/job_endpoint.go

```diff
@@ -516,3 +491,155 @@ func validateJob(job *structs.Job) error {
 
 	return validationErrors.ErrorOrNil()
 }
+
+// Status returns a summary of the status of the job's allocations (how many
+// pending, running, complete, failed).
+func (j *Job) Status(args *structs.JobSpecificRequest,
+	reply *structs.JobStatusResponse) error {
+	if done, err := j.srv.forward("Job.Status", args, args, reply); done {
+		return err
+	}
+	defer metrics.MeasureSince([]string{"nomad", "job", "status"}, time.Now())
+
+	snap, err := j.srv.fsm.State().Snapshot()
+	if err != nil {
+		return err
+	}
+
+	out, err := snap.JobByID(args.JobID)
+	if err != nil {
+		return err
+	}
+
+	if out == nil {
+		return fmt.Errorf("job %q not found", args.JobID)
+	}
+
+	status, err := j.computeJobStatus(snap, out)
+	if err != nil {
+		return err
+	}
+
+	*reply = *status
+	reply.Index = out.ModifyIndex
+
+	// Set the query response
+	j.srv.setQueryMeta(&reply.QueryMeta)
+	return nil
+}
+
+// computeJobStatus takes a state snapshot and the job and computes is high
+// level status
+func (j *Job) computeJobStatus(snap *state.StateSnapshot, job *structs.Job) (*structs.JobStatusResponse, error) {
+	if job == nil {
+		return nil, fmt.Errorf("job can not be nil")
+	}
+
+	status := &structs.JobStatusResponse{
+		Status:     job.Status,
+		TaskGroups: make(map[string]structs.AllocStateCounts, len(job.TaskGroups)),
+	}
+	allocs, err := snap.AllocsByJob(job.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Periodic jobs can't have running instances
+	if job.IsPeriodic() {
```

Comment:
I am not sure that is desirable? The important thing on periodic jobs is the list of running of children jobs